addons/l10n_in/models/company.py

```diff
@@ -12,3 +11,44 @@ class ResCompany(models.Model):
         ],
         string="HSN Code Digit",
     )
+
+    def write(self, vals):
+        vals_state = vals.get('state_id')
+        flag = []
+        for company in self:
+            if vals_state and not company._origin.state_id:
+                flag.append(company.name)
+        res = super(ResCompany, self).write(vals)
+        for company in self:
+            if company.name in flag:
+                company.update_fiscal_position()
+        return res
+
+    def _get_fiscal_vals(self):
+        self.ensure_one()
+        indian_all_states = self.env['res.country.state'].search([('country_id.code', '=', 'IN')])
+        return {
+            'auto_apply': True,
+            'country_id': self.country_id.id,
+            'state_ids': [Command.set((indian_all_states - self.state_id).ids)]
```

Comment:
I think you can avoid this.  Why not use the inherit of the load_data of the CoA to add a fiscal position in auto_apply with a sequence that comes first and the state_id == the state of the company (=within the state, intrastate)) and then the second fiscal position will always just have country India and be called interstate (and we already have it).  @jva-odoo

Reply:
Thanks for your input @JCO

The original object of this task was as follows :

upon installation of l10n_in
1. Inter State fiscal position get created automatically ( as working now )
2. addition to above  - Country = company country, state = except company state,
3. Auto detect = True.

Reply:
i guess it's the same as you guided by code. we exactly expecting the same,
i guess jva gave it another height that to cover advance cases,
currently we recommend user to manage advance cases manually.

Reply:
Yes, but you do not have the complexity of adding all other states.  You just add the state it is in and you apply the fiscal position before the others.  You can actually quickly configure in a db to try.